### PR TITLE
Integrate Dashboard (Listings Page) UI with server

### LINKED ISSUE #37: IN: Listings Page

### PR DESCRIPTION
**Pull request by EdwardOlmos**
### What this PR does (required):
- Resolves #37
- Prerequisite: #21 
- Allows users to search for pet sitters by location on the dashboard for a profile listings view
- Modifies Profile Model, Controller, and Route
- Modifies Dashboard UI
- Add Client API Helper to fetch pet sitter account type with a valid location and price

### Screenshots / Videos (front-end only):
- Navigate to Dashboard (/dashboard)
profile_listings

### Any information needed to test this feature (required):
Create users and modify every user's profile:

1. Update to pet_sitter
2. Update Rating
3. Update Pay
4. Update Address
5. Update Description
6. Update Headline
7. Update Photo (Use Unsplash Photos)

Go to the dashboard and search with an empty string or location

### Any issues with the current functionality (optional):
- I'm not sure how to approach the 'show more' functionality

### CODE REVIEW

File: server/controllers/profile.js [hunk 1]
```diff
@@ -38,4 +36,36 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
       profile: profile,
     },
   });
-});
\ No newline at end of file
+});
+
+// @route GET /profile
+// @desc Get list of pet sitter profiles by location or get all
+// @access Public
```
Comment (ezeilo-su) [inline server/controllers/profile.js:43]: Good documentation

**Review by ezeilo-su — COMMENTED** [review]
Good work with this one!

File: client/src/components/Cards/ProfileCard.tsx [hunk 2]
```diff
@@ -29,46 +30,40 @@ const useStyles = makeStyles({
...
 
-const ProfileCard: React.FC<ProfileCardProps> = ({ photo, name, caption, rating, description, location, pay }) => {
+const ProfileCard: React.FC<ProfileCardProps> = ({ profile }) => {
   const classes = useStyles();
   return (
     <Card className={classes.root}>
       <CardActionArea>
         <CardContent sx={{ display: 'flex', flexDirection: 'column', justifyItems: 'center' }}>
-          <Avatar alt="Profile Photo" src={photo} className={classes.avatar} />
+          <Avatar alt="Profile Photo" src={profile?.photo} className={classes.avatar} />
```
Comment (rajivtitus) [inline client/src/components/Cards/ProfileCard.tsx:42]: As opposed to using ternary operators before accessing object properties, why don't we consider rendering the `ProfileCard ` component only if a profile exists 😄

File: server/controllers/profile.js [hunk 3]
```diff
@@ -38,4 +36,36 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
...
+
+    const profiles = await Profile.find({ accountType: 'pet_sitter', address: { $regex: req.query.location, $options: "i" }, pay: { $ne: null } });
+
+    res.status(200).json({
+      profiles: profiles
+    });
+
+  } else if (req.query.location === '') {
+
+    const profiles = await Profile.find({ accountType: 'pet_sitter', address: { $ne: null | '' }, pay: { $ne: null } });
```
Comment (rajivtitus) [inline server/controllers/profile.js:56]: Both queries are almost identical, why don't we combine them? We can store the query filter in a variable based on whether query.location exists or not

**Review by rajivtitus — COMMENTED** [review]
Good work! Just a couple of minor comments to address :)

File: client/src/components/Cards/ProfileCard.tsx (hunk 2 above)
Comment (EdwardOlmos) [reply client/src/components/Cards/ProfileCard.tsx]: I removed the ternary operators. I believe ProfileCard won't render unless a Profile is passed in through props

**Pull request merged**